health_vara.py

```diff
@@ -64,9 +70,24 @@ def search_mobile(cls, _name, clause):
         value = clause[2]
 
         res.append(('contact_mechanisms.type', '=', 'mobile'))
-        res.append(('contact_mechanisms.value', op, value))
+        res.append(('contact_mechanisms.rec_name', op, value))
         return res
 
+    @classmethod
+    def search_rec_name(cls, name, clause):
+        if clause[1].startswith('!') or clause[1].startswith('not '):
+            bool_op = 'AND'
+        else:
+            bool_op = 'OR'
+
+        base = super().search_rec_name(name, clause)
+
+        # GnuHealth module removes the contact mechanisms from the global search
+        # This adds it back in again, ontop of the usual logic
+        return [bool_op,
+                base,
+                ('contact_mechanisms.rec_name',) + tuple(clause[1:]),]
```

Comment:
Makes the global search work for mobile phone numbers without spaces:
![image](https://github.com/vara-ai/health_vara/assets/1367955/f2d6a91a-21c9-4e33-a491-f654f12322cc)